builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -327,7 +326,39 @@ func resourceAwsElasticBeanstalkEnvironmentUpdate(d *schema.ResourceData, meta i
 		os := o.(*schema.Set)
 		ns := n.(*schema.Set)
 
-		updateOpts.OptionSettings = extractOptionSettings(ns.Difference(os))
+		rm := extractOptionSettings(os.Difference(ns))
+		add := extractOptionSettings(ns.Difference(os))
+
+		// Additions and removals of options are done in a single API call, so we
+		// can't do our normal "remove these" and then later "add these", re-adding
+		// any updated settings.
+		// Because of this, we need to remove any settings in the "removable"
+		// settings that are also found in the "add" settings, otherwise they
+		// conflict. Here we loop through all the initial removables from the set
+		// difference, and we build up a slice of settings not found in the "add"
+		// set
+		var remove []*elasticbeanstalk.ConfigurationOptionSetting
+		if len(add) > 0 {
+			for _, r := range rm {
+				for _, a := range add {
+					if *r.Namespace == *a.Namespace && *r.OptionName == *a.OptionName {
```

Comment:
In the statefile, we ignore `ResourceName` unless it's a scheduled action. I think here I'll check if the local setting has `ResourceName` and then check if they match, continuing if they do not match. If we don't have it set locally, I'll just continue on as we have here. Does that sound reasonable?